tests/Unit/NumericalAlgorithms/Spectral/Python/Test_Mesh.py

```diff
@@ -112,6 +112,79 @@ def test_nonuniform_all_with_pickle(self):
             self.check_quadrature(m2, quadratures2D)
             self.check_quadrature(m3, quadratures3D)
 
+    def test_equality(self):
+        for basis in self.bases:
+            for quadrature in self.quadratures:
+                for i in range(100):
```

Comment:
Seems to me these tests would be a lot simpler if you loop over the three dimensions, then you don't have to write everything three times. I don't think you necessarily need to repeat the random test 100 times, a single iteration should suffice.